Fix IPV6 networking

Mathyn opened the pull request:
When hosting or joining a server via a IPV6 address no connection can be made. This is because of the ipv6 address '::' which should be '::1'. This PR fixes this mistake.

DanielGibson commented:
I don't think this is correct.

`::1` is the equivalent of `127.0.0.1`, so it's a loopback-address, only reachable from the same machine, see https://en.wikipedia.org/wiki/Localhost

`::` is the equivalent of `0.0.0.0`, so it's the unspecified address, and listening on that address means "accept connections on all network interfaces and all their addresses", see https://en.wikipedia.org/wiki/IPv6_address#Unspecified_address

So it should be correct to listen on `::` (and not `::1`) here.

Mathyn commented:
You are correct! I changed it to '::/0'. For some reason this address does work and simply doing '::' caused 'address in use' errors on the dedicated server and simply finding no servers on the client.

Mathyn commented:
Just to clarify: '::' does not work on MacOS. I have not tested on Windows systems which might understand '::' to mean '::/0'. 

Maybe this is what caused this issue: #263 ?

DanielGibson commented:
I don't think this caused #263, after all that one was about IPv4?

Really weird that MacOS apparently needs `::/0` instead of `::`

Mathyn commented:
I'm not sure why it doesn't work. '::' should work but every time we use '::' no servers can be found and the dedicated server claims the address is already in use.

So far I've been able to verify this behavior on three Macbooks.

DanielGibson commented:
What are the exact error messages you get from the game?

And if I understand https://en.wikipedia.org/wiki/IPv6_address#Special_addresses correctly, it should probably be `::/128`, not `::/0`? Does that work as well?

Mathyn commented:
When using address '::' the following error is shown when trying to start a dedicated server:

```
NET_Socket: bind: Address already in use
```

When trying to join a server from the client this error is shown twice. No other errors are displayed as far as I can see.

Changing the address to '::/128' also works and looking at the table you posted appears to be the correct address. I'll update the PR.

This is really strange. I'm assuming (but have not tested) using address '::' works just fine on Windows and Linux?

Just to be complete. This is the terminal output when starting a dedicated server with the address set to '::'.

```
Yamagi Quake II v7.31pre
========================
Platform: Darwin
Architecture: x86_64
Byte ordering: little endian

Added packfile './baseq2/pak0.pak' (3308 files).
Added packfile './baseq2/pak1.pak' (279 files).
Added packfile './baseq2/pak2.pak' (2 files).
Added packfile '/Users/mathyn/Games/quake2/.//baseq2/pak0.pak' (3308 files).
Added packfile '/Users/mathyn/Games/quake2/.//baseq2/pak1.pak' (279 files).
Added packfile '/Users/mathyn/Games/quake2/.//baseq2/pak2.pak' (2 files).
Using '/Users/mathyn/.yq2//baseq2' for writing.
execing default.cfg
couldn't exec yq2.cfg
execing config.cfg
couldn't exec autoexec.cfg
NET_Socket: bind: Address already in use
-------- game initialization -------
LoadLibrary("game.dylib")
Game is starting up.
Game is baseq2 built on Sep 11 2018.
------------------------------------

------- server initialization ------
50 entities inhibited.
1 teams with 2 entities.
------------------------------------

==== Yamagi Quake II Initialized ====

*************************************
```

And this is the output shown in the terminal when entering the 'join server' menu in the client:

```
NET_Socket: bind: Address already in use
NET_Socket: bind: Address already in use
pinging broadcast...
pinging multicast...
NET_SendPacket: IPv6 multicast destination but +set multicast not specified: ff12::666
pinging 192.168.3.51..
```

When trying to host a game from the client I get the following output:

```
maxclients will be changed for next game.
deathmatch will be changed for next game.
==== ShutdownGame ====
NET_Socket: bind: Address already in use
NET_Socket: bind: Address already in use
-------- game initialization -------
LoadLibrary("game.dylib")
Game is starting up.
Game is baseq2 built on Sep 11 2018.
------------------------------------

------- server initialization ------
0 entities inhibited.
0 teams with 0 entities.
------------------------------------

Server restarted
Server disconnected, reconnecting
Connecting to localhost...
[0.0.0.0]:0: challenge
Mathyn connected
[0.0.0.0]:0: client_connect


-===================================-

 The Edge
```

Mathyn commented:
Any updates on this? Let me know if I need to investigate more or make any changes :)

Yamagi commented:
And merged, thank you :)
